automated-comments/javascript/gigasecond/prefer_extracted_top_level_constant.md

```diff
@@ -0,0 +1,9 @@
+Instead of defining the constant _inside_ the function, consider extracting it
```

Comment:
Should this perhaps be a general comment, where the name of exported value is parameterized?

Reply:
Perhaps, but this is sorta a special case 😅. I'll move it to general if this comes up more often.

Basically, usually I don't just say to "extract it", but "extract it because xyz". In the case of `gigasecond`, most of the time people have a literal or big number comprehension inline. In that case it uses a comment that's already in `website-copy` that explains you can extract it into a `const`. This comment is for when it's already a `const`,  but defined inside the function body. In the case of `resistor-color`, when people define it inside the function body, they re-create the colors array all the time, which is the extra commentary given there.

In other words, I think in general when we're saying to extract it to the top level, there are extra reasons why, **except** for this particular case.

Reply:
Right, that makes sense.